addons/purchase_stock/models/product.py

```diff
@@ -140,14 +135,39 @@ def _compute_monthly_demand(self):
 
     @api.model
     def _get_monthly_demand_moves_location_domain(self):
-        return Domain.OR([
+        """ Returns all moves coming from the selected warehouse that are:
+                - going to customer locations or used in production
+                - going to other warehouses (eg. central warehouse dispatching to stores)"""
+
+        warehouse_id = self.env.context.get('warehouse_id', False)
+        from_selected_warehouse, to_other_warehouse = Domain.TRUE, Domain.TRUE
+        internal_transit_location_id = self.env.companies.internal_transit_location_id.ids
```

Comment:
Can be an issue if you are searching for a warehouse that is not in your current company. I would rather do 

```suggestion
        internal_transit_location_id = warehouse.company_id.internal_transit_location_id.ids
```